ballerina/Package.md

```diff
@@ -1,18 +1,24 @@
 ## Package Overview
 
-The `oauth2` library is one of the standard library modules of the <a target="_blank" href="https://ballerina.io/">Ballerina</a> language.
-
-This module provides a framework for interacting with OAuth2 authorization servers as specified in the <a target="_blank" href="https://datatracker.ietf.org/doc/html/rfc6749">RFC 6749</a> and <a target="_blank" href="https://datatracker.ietf.org/doc/html/rfc7662">RFC 7662</a>.
+This package provides a framework for interacting with OAuth2 authorization servers as specified in the [RFC 6749](https://datatracker.ietf.org/doc/html/rfc6749) and [RFC 7662](https://datatracker.ietf.org/doc/html/rfc7662).
 
 The OAuth 2.0 authorization framework enables a third-party application to obtain limited access to an HTTP service either on behalf of a resource owner by orchestrating an approval interaction between the resource owner and the HTTP service or by allowing the third-party application to obtain access on its own behalf.
 
-The Ballerina `oauth2` module facilitates auth providers that are to be used by the clients and listeners of different protocol connectors.
+The Ballerina `oauth2` package facilitates auth providers that are to be used by the clients and listeners of different protocol connectors.
+
+### Listener OAuth2 Provider
+
+Represents the listener OAuth2 provider, which is used to validate the received credential (access token) by calling the configured OAuth2 introspection endpoint.
+
+### Client OAuth2 Provider
+
+Represents the client OAuth2 provider, which is used to generate OAuth2 access tokens using the configured OAuth2 token endpoint configurations. This supports the client credentials grant type, password grant type, and refresh token grant type.
 
-### Report Issues
+## Report Issues
 
-To report bugs, request new features, start new discussions, view project boards, etc., go to the <a target="_blank" href="https://github.com/ballerina-platform/ballerina-standard-library">Ballerina standard library parent repository</a>.
+To report bugs, request new features, start new discussions, view project boards, etc., go to the [Ballerina standard library parent repository](https://github.com/ballerina-platform/ballerina-standard-library).
 
-### Useful Links
+## Useful Links
```

Comment:
Don't we need the Docs and BBE links?

Reply:
We only have these in the `README.md`, but not in the `Package.md`. Right? If it is needed, I will add those to `Package.md` as well.

Reply:
Yeah, we don't need it. We can add those later as central evolves.

Reply:
Ack.